virtualpdu/core.py

```diff
@@ -42,13 +43,25 @@ def pdu_outlet_state_changed(self, pdu, outlet, state):
         except KeyError:
             return
 
+        self._async_change(device, state)
+
+    def _async_change(self, device, state):
         if state == POWER_ON:
-            self.driver.power_on(device)
+            def switch_power():
+                self.driver.power_on(device)
         elif state == POWER_OFF:
-            self.driver.power_off(device)
+            def switch_power():
+                self.driver.power_off(device)
         elif state == REBOOT:
-            self.driver.power_off(device)
-            self.driver.power_on(device)
+            def switch_power():
+                self.driver.power_off(device)
+                self.driver.power_on(device)
+        else:
+            self.logger.error("Unknown power state: {}".format(state))
+            return
+
+        with ThreadPoolExecutor(max_workers=1) as executor:
```

Comment:
Interesting point. I honestly went with the family when I implemented ThreadPoolExecutor